processing/src/main/java/io/druid/query/aggregation/JavaScriptAggregatorFactory.java

```diff
@@ -40,7 +40,7 @@
 import java.util.Comparator;
 import java.util.List;
 
-public class JavaScriptAggregatorFactory implements AggregatorFactory
+public class JavaScriptAggregatorFactory extends AggregatorFactory
```

Comment:
i dont think we can guarantee any order in all possible [auto] merging situations. i think, we can override to string match fnAggregate, fnReset and fnCombine, fail if they don't match.
As far as schema evolution is considered, user always has the option to explicitly supply list of aggregators (that is why I added same in AppendTask too).